pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -939,7 +901,16 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 	}
 
 	if _, needsRemediation := machine.Annotations[externalRemediationAnnotation]; !needsRemediation {
-		return nil
+		return a.removeNodeFinalizer(ctx, machine)
```

Comment:
just out of curiosity - why finalizer removal is here and not in Update/Create?
and why only if the machine doesn't need remediation?
looks like we're not using finalizer anymore, so we don't care to remove it in all cases